### ISSUE TITLE
vmalert: allow to blackhole alerting notifications

### ISSUE THREAD

**Issue by wf1nder** [VictoriaMetrics/VictoriaMetrics#4122]
### Describe the solution you'd like

We want to start using vmalert, and we have our own implementation of the component that processing and escalating alerts, instead of using alertmanager, eg.
This component uses `/api/v1/alerts` handler to get information about all alerts from prometheus or promxy. I see that vmalert also providing this handler, so looks like we can just replace our promxy with vmalert in this case.
So for our purposes there is no need to sent notifications to alertmanager or something. But vmalert refuses to start without any notification target:
```
fatal	VictoriaMetrics/app/vmalert/main.go:143	failed to start: config contains alerting rules but neither `-notifier.url` nor `-notifier.config` aren't set
```

Now I specified dummy target like `-notifier.url=http://127.0.0.1:1234`, and vmalert successfully started and working, and just sometimes throws errors:
```
failed to send alerts to addr "http://127.0.0.1:1234/api/v2/alerts": Post "http://127.0.0.1:1234/api/v2/alerts": dial tcp 127.0.0.1:1234: connect: connection refused
```

But it would be nice to make options `-notifier.url` and `-notifier.config` optional.

**Comment by hagen1778** [comment]
The message vmalert throws is supposed to notify user about misconfiguration: it would be very not cool to expect for alerts when notifier is not specified. And with proposed solution I see no way to differentiate case when notifier is important and when it is not.
I see multiple ways how to solve this, but I don't thing either of these ways is helpful for vmalert users without custom solution like yours.

**Comment by wf1nder** [comment]
> The message vmalert throws is supposed to notify user about misconfiguration

I have nothing against notification, but vmalert throws fatal here, and then exiting with non-zero exit code.
Do you think using a fake address is the correct way? I think it's an ugly crutch.

As an alternative solution, I suggest to add the ability to explicitly disable sending notifications. Something like `-no-notifier` flag.

**Comment by hagen1778** [comment]
> I have nothing against notification, but vmalert throws fatal here, and then exiting with non-zero exit code.
Do you think using a fake address is the correct way? 

The fatal error is needed to notify user that core functionality of the tool can't be conducted because of misconfiguration. Otherwise, user may miss the normal log message and continue using misconfigured tool for unknown period of time.

> I think it's an ugly crutch.

We should remember that vmalert isn't used here for what it was designed for. Your scenario of use is pretty unique and is unlikely to be needed by someone else.

> As an alternative solution, I suggest to add the ability to explicitly disable sending notifications. Something like -no-notifier flag.

I think, we might have something like `notifier.blackhole` cmd-line flag, which would work as a real notifier except doing any work. This flag might be useful for local testing, to simplify the procedure of running vmalert.

**Comment by venkatbvc** [comment]
@hagen1778  when notifier.blackhole is enabled, then vmalert doen't send any notifications? and fatal notifier.url check would be skipped?  is my understanding right.
i would like to contribute

**Comment by hagen1778** [comment]
> then vmalert doen't send any notifications? and fatal notifier.url check would be skipped? is my understanding right.

in my understanding the notifier.blackhole won't check if notifier.URL was set and on notifier.Init() func it will return a no-op mock. With this, vmalert may continue working as before, sending alerts and resolves. But no actual communication with external alertmanager will happen - all calls to Send() function will be no-op.

**Comment by venkatbvc** [comment]
@hagen1778  Thanks. I will send a PR

**Comment by venkatbvc** [comment]
@hagen1778  sent a PR via https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4639. Please review and let me know if any changes are required

**Comment by valyala** [comment]
The `-notifier.blackhole` command-line flag is available in `vmalert` starting from [v1.92.0](https://docs.victoriametrics.com/CHANGELOG.html). If this flag is specified, then alerts aren't sent to any external notifiers.

Closing this feature request as done.
